tests/unit/test_discovery_client.py

```diff
@@ -249,6 +254,51 @@ def test___get_discovery_service_address___key_file_not_exist___throws_timeouter
     assert exc_info.type is TimeoutError
 
 
+def test___get_discovery_service_address___open_key_file_throws_windowserror___throws_timeouterror(
+    mocker: MockerFixture,
+    temp_discovery_key_file_path: pathlib.Path,
+    temp_registration_json_file_path: pathlib.Path,
+):
+    if sys.platform != "win32":
+        pytest.skip(f"Platform {sys.platform} is not supported")
+    else:
+        mocker.patch(
+            "ni_measurementlink_service._internal.discovery_client._get_key_file_path",
+            return_value=temp_discovery_key_file_path,
+        )
+        mocker.patch(
+            "ni_measurementlink_service._internal.discovery_client._START_SERVICE_TIMEOUT", 5.0
+        )
+        mocker.patch(
+            "ni_measurementlink_service._internal.discovery_client._open_key_file",
+            side_effect=WindowsError,
+        )
+        mocker.patch(
+            "ni_measurementlink_service._internal.discovery_client._get_registration_json_file_path",
+            return_value=temp_registration_json_file_path,
+        )
+        mocker.patch("subprocess.Popen")
+
+        with pytest.raises(IOError) as exc_info:
+            _get_discovery_service_address()
+        assert exc_info.type is TimeoutError
+
+
+def test___key_file_not_exist___open_key_file___raises_file_not_found_error(
+    mocker: MockerFixture, temp_discovery_key_file_path: pathlib.Path
+):
+    if sys.platform != "win32":
+        pytest.skip(f"Platform {sys.platform} is not supported")
```

Comment:
Use the decorator: `@pytest.mark.skipif(sys.platform != "win32", reason="Windows-only test")`